src/avram/join.cr

```diff
@@ -4,7 +4,14 @@ module Avram::Join
   abstract class SqlClause
     getter :from, :to, :from_column, :to_column
 
-    def initialize(@from : Symbol, @to : Symbol, @primary_key : Symbol? = nil, @foreign_key : Symbol? = nil, @comparison : String? = "=", @using : Array(Symbol) = [] of Symbol)
+    def initialize(
+      @from : Symbol,
+      @to : Symbol,
+      @primary_key : Symbol | String | Nil = nil,
```

Comment:
Not sure if this is related, but in the model you set the `primary_key_name` to return Symbol specifically; should we look at typing this to `Symbol?` instead?

Reply:
Yeah that makes sense!